be/src/olap/tablet.cpp

```diff
@@ -1785,7 +1785,7 @@ Status Tablet::prepare_compaction_and_calculate_permits(CompactionType compactio
         if (!res.ok()) {
             set_last_base_compaction_failure_time(UnixMillis());
             *permits = 0;
-            if (!res.is<BE_NO_SUITABLE_VERSION>()) {
+            if (!res.is<BE_NO_SUITABLE_VERSION>() && !res.is<FULL_NO_SUITABLE_VERSION>()) {
```

Comment:
don't change here, this is error code for base compaction, and `FULL_NO_SUITABLE_VERSION` is generated by full compaction.